primitives/src/block.rs

```diff
@@ -269,6 +279,139 @@ impl Encodable for Header {
     }
 }
 
+/// The decoder for the [`Header`] type.
+#[cfg(feature = "alloc")]
+pub struct HeaderDecoder(
+    Decoder6<
+        VersionDecoder,
+        BlockHashDecoder,
+        TxMerkleNodeDecoder,
+        BlockTimeDecoder,
+        CompactTargetDecoder,
+        encoding::ArrayDecoder<4>, // Nonce
```

Comment:
Its a public holiday here but I wanted to get this PR rebased, but I've only half managed to context switch into this work. Flagging that we probably want a `NonceDecoder` here.